livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamChannels.kt

```diff
@@ -2,163 +2,141 @@ package io.getstream.livestream.compose.ui
- * @param columnCount - a toggle State to switch between 2 column grid to single column list
+ * @param isGrid - a toggle State to switch between 2 column grid to single column list
+ * @param gridColumn - customization for Grid view column count.
  */
 @ExperimentalMaterialApi
 @ExperimentalFoundationApi
 @Composable
 fun LiveStreamChannels(
-    channelListViewModel: ChannelListViewModel,
     modifier: Modifier = Modifier,
+    channelListViewModel: ChannelListViewModel = viewModel(
+        factory =
+        ChannelViewModelFactory(
+            ChatClient.instance(),
+            ChatDomain.instance(),
+            QuerySort.desc("last_updated"),
+            Filters.and(
+                Filters.eq("type", "livestream"),
+                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: ""))
+            )
+        )
+    ),
     isDarkTheme: Boolean = false,
-    columnCount: Int = 2
+    isGrid: Boolean = false,
+    gridColumn: Int = 2
 ) {
-    LaunchedEffect(Unit) {
-        channelListViewModel.start()
-    }
-    val cardBackground = if (isDarkTheme)
-        colorResource(id = R.color.cardview_dark_background)
-    else
-        colorResource(id = R.color.cardview_light_background)
     val context: Context = LocalContext.current
 
-    val channels = channelListViewModel.channelsState.channels.map { channel ->
-        LiveStreamChannelItem(
-            channelId = channel.cid,
-            channelArt = context.randomArtWork(),
-            channelName = channel.name,
-            channelDescription = context.randomDescription()
-        )
-    }
+    // We load a [LazyVerticalGrid] when we have a grid view to show
+    if (isGrid) {
+        // We fetch channels first
+        val channels = channelListViewModel.channelsState.channels.map { channel ->
+            channel.toLiveStreamChannelItem(context = context)
+        }
 
-    LazyVerticalGrid(
-        modifier = modifier,
-        cells = GridCells.Fixed(columnCount),
-        contentPadding = PaddingValues(bottom = 12.dp, top = 12.dp),
-        content = {
-            items(channelListViewModel.channelsState.channels.size) { index ->
-                Card(
-                    shape = RoundedCornerShape(8.dp),
-                    backgroundColor = cardBackground,
-                    modifier = Modifier
-                        .fillMaxWidth()
-                        .padding(10.dp)
-                        .clickable(
-                            interactionSource = remember { MutableInteractionSource() },
-                            indication = rememberRipple(bounded = true),
-                            onClick = {
-                                openMessages(context, channels[index])
-                            }
-                        ),
-                    elevation = 4.dp,
-                ) {
-                    Column(
+        LazyVerticalGrid(
+            modifier = modifier.background(ChatTheme.colors.appBackground),
```

Comment:
Should we be adding this hardcoded background here? Or passing it in from the outside?